File: coding/python/armstrong.py
n = int(input())
l = len(str(n))
n1 = n
num = 0
while(n1 != 0):
    digit = n1 % 10
    num += digit**l
    n1 //= 10

if num == n:
    print("It is an Armstrong Number!!")
else:
    print("It is not an Armstrong Number!!")

File: coding/C/armstrong.c
#include <stdio.h>
#include <math.h>

int length(int n){   
    int i = 0;
    while(n != 0){
        i++;
        n /= 10;
    }
    return i;
}

int armstrong(int n, int l){
    int sum = 0;
    for(int i  = 0; i < l; i++){
        int r = n % 10;
        sum += pow(r, l);
        n /= 10;
    }
    return sum;
}

int main(){
    int n, num;
    scanf("%d", &n);

    if(n == armstrong(n, length(n))){
        printf("Armstrong!");
    } else {
        printf("Not Armstrong!");
    }
}